utils.go

```diff
@@ -16,18 +17,28 @@ const (
 	TagsHTTPHeaderPrefix = "Open-Tracing-Trace-Tags-"
 )
 
-// PropagateSpanInHeader encodes Span `ctx` to `h` as a series of
-// HTTP headers. Values are URL-escaped.
-func PropagateSpanInHeader(
+// InjectSpanInHeader encodes Span `sp` in `h` as a series of HTTP headers.
+// Values are URL-escaped.
+func InjectSpanInHeader(
 	sp Span,
 	h http.Header,
-	propagator SpanPropagator,
-) {
-	contextIDMap, tagsMap := propagator.PropagateSpanAsText(sp)
-	for headerSuffix, val := range contextIDMap {
+) error {
+	// First, look for a PROPAGATION_FORMAT_GO_HTTP_HEADER injector (our preference).
+	injector := sp.PropagationInjectorForFormat(PROPAGATION_FORMAT_GO_HTTP_HEADER)
```

Comment:
@tschottdorf @yurishkuro please read this function and the one below... the idea is that propagation code could first check for its format of choice (whatever that may be), then fall back on something like `PROPAGATION_FORMAT_SPLIT_TEXT` if the OpenTracing impl doesn't know about the aforementioned "format of choice".